pkg/skaffold/deploy/labels.go

```diff
@@ -48,7 +49,11 @@ type Labeller interface {
 }
 
 // merge merges the labels from multiple sources.
-func merge(deployer Labeller, sources ...Labeller) map[string]string {
+func merge(opts config.SkaffoldOptions, deployer Labeller, sources ...Labeller) map[string]string {
+	if opts.NoLabels {
```

Comment:
we don't need to pass the entire `SkaffoldOptions` here. just passing the `NoLabels` bool should be enough